app/models/shopping_list_item_builder.rb

```diff
@@ -1,59 +1,45 @@
 # frozen_string_literal: true
 
 class ShoppingListItemBuilder
-  attr_reader :ingredients, :meal_plan, :recipe
+  attr_reader :shopping_list, :ingredients
 
-  def initialize(shopping_list:, single_ingredient: nil, meal_plan:, recipe:)
-    @shopping_list = shopping_list
-    @single_ingredient = single_ingredient
-    @meal_plan = meal_plan
-    @recipe = recipe
-    @ingredients = assign_ingredients
+  def initialize(shopping_list_id:, ingredient_ids:)
+    @shopping_list = ShoppingList.find_by(id: shopping_list_id)
+    @ingredients = Ingredient.where(id: ingredient_ids)
   end
 
+  def add_items_to_list
+    self.ingredients.each do |ingredient|
+      add_ingredient_to_list(ingredient)
+    end
+  end
+
+  private
+
   def add_ingredient_to_list(ingredient)
-    item_name = "#{ingredient.measurement_unit} #{ingredient.name}"
-    item = @shopping_list.shopping_list_items.find_by(name: item_name)
+    item_on_list = self.shopping_list.shopping_list_items.find_by(name: ingredient.measurement_and_name)
```

Comment:
This rename is an attempt at more clarity, but dunno if you like that.

Reply:
Works for me.